lib/elixir/lib/uri.ex

```diff
@@ -265,10 +265,11 @@ defmodule URI do
   See `encode_www_form` if you are interested in escaping reserved
   characters too.
 
-  This function also accepts a `predicate` function as an optional
-  argument. If passed, this function will be called with each byte
-  in `string` as its argument and should return `true` if the given
-  byte should be left as is.
+  It also accepts a function as an optional argument. If passed, `fun` will be
```

Comment:
```suggestion
  This function also accepts a predicate function as an optional argument. If passed, `fun` will be
```

Reply:
Well, I also like that, but I had to change it, because it doesn't accept a predicate function, but any function. That's what this PR fixes. If we want to stick to the predicate function we need to error when no boolean is returned then.

Reply:
See my comment bellow for my answer
https://github.com/elixir-lang/elixir/pull/8779#discussion_r254778179